lib/src/Base/Stat/CorrelationAnalysis.cxx

```diff
@@ -61,9 +61,10 @@ Point CorrelationAnalysis::SpearmanCorrelation(const Sample & inputSample,
 Point CorrelationAnalysis::SRC(const Sample & inputSample,
                                const Sample & outputSample)
 {
+  const UnsignedInteger dimension = inputSample.getDimension();
+  if (dimension < 2) throw InvalidDimensionException(HERE) << "Error: input sample must have dimension > 1";
   if (outputSample.getDimension() != 1) throw InvalidDimensionException(HERE) << "Error: output sample must be 1D";
   if (inputSample.getSize() != outputSample.getSize()) throw InvalidArgumentException(HERE) << "Error: input and output samples must have the same size";
-  const UnsignedInteger dimension = inputSample.getDimension();
   // Var(X+a) = Var(X); However for numerical stability, data are centered
```

Comment:
@sofianehaddad Excellent idea. IMO it should be done in `LinearLeastSquares`, instead of taking `center` equal to zero in the `build()` method.

Reply:
@regislebrun IMO several things should be updated in `LinearLeastSquares`. A specific PR will be done soon